backend/controllers/authController.js

```diff
@@ -22,129 +20,106 @@ const verifyToken = async (req, res, next) => {
+    );
+
+    // Set new tokens in cookies
+    res.cookie("token", newToken, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === 'production',
+    });
+    res.cookie("refreshToken", newRefreshToken, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === 'production',
+    });
+
+    return res.status(200).json({
+      status: "success",
+      message: "Token refreshed successfully",
+    });
+  } catch (error) {
+    return res.status(401).json({
+      status: "fail",
+      message: "Invalid refresh token",
+    });
+  }
 };
 
-// Signup controller
+// signup controller
 const signup = async (req, res, next) => {
-  const { username, email, password, confirmPassword, firstName, lastName, contact } = req.body;
+  const body = req.body;
+  console.log(body);
 
-  if (password.length < 7) {
-      return res.status(400).json({
-          status: 'fail',
-          message: 'Password must be at least 7 characters long',
-      });
+  // password length validation
+  if (body.password.length < 7) {
+    return res.status(400).json({
+      status: "fail",
+      message: "Password must be at least 7 characters long",
+    });
   }
+  // attempt to create user in db (unsuccessful if doesn't pass validations)
+  try {
+    const newUser = await db.User.create({
+      username: body.username,
+      email: body.email,
+      password: body.password,
+      confirmPassword: body.confirmPassword,
+    });
 
-  if (password !== confirmPassword) {
-      return res.status(400).json({
-          status: 'fail',
-          message: 'Passwords do not match',
-      });
-  }
+    const result = newUser.toJSON();
+    delete result.password;
+    delete result.deletedAt;
 
-  try {
-      const newUser = await db.User.create({
-          username,
-          email,
-          password,
-          contact,
-          firstName,
-          lastName
-      });
+    result.token = generateToken({
+      id: result.id,
+    });
 
-      const result = newUser.toJSON();
-      delete result.password;
-
-      const token = generateToken(
-        {id: result.id},
-        process.env.JWT_SECRET,
-        process.env.JWT_EXPIRES_IN
-      );
-      const refreshToken = generateToken(
-        { id: result.id },
-        process.env.JWT_SECRET,
-        process.env.JWT_REFRESH_EXPIRES_IN
-      );
-
-      res.cookie('token', token, { httpOnly: true, secure: process.env.NODE_ENV === 'production' });
-      res.cookie('refreshToken', refreshToken, { httpOnly: true, secure: process.env.NODE_ENV === 'production' });
-
-      return res.status(201).json({
-          status: 'success',
-          message: 'User created successfully',
-          data: result,
+    if (!result) {
+      return res.status(400).json({
+        status: "fail",
+        message: "Failed to create user",
       });
+    }
+
+    return res.status(201).json({
+      status: "success",
+      message: "User created successfully",
+      data: result,
+    });
```

Comment:
```suggestion
    return res.status(201).json({
      success: true/false,
      message: "User created successfully",
      data: result,
    });
```
we already have status as 201, which is a standard for created resources
success could be a boolean

Reply:
I would leave status, to follow standard messaging in the whole project